src/mongocxx/database.hpp

```diff
@@ -248,6 +248,7 @@ class MONGOCXX_API database {
 
     class MONGOCXX_PRIVATE impl;
     std::unique_ptr<impl> _impl;
+    const std::unique_ptr<impl>& get_impl() const;
```

Comment:
The most logically correct approach would be to provide a const getter and a separate non-const getter.   Also, it seems strange to return any sort of unique_ptr, I'd rather we just returned a reference or bare ptr to the underlying _impl:
```
// const getter
const impl& get_impl() const;
// non-const getter
impl* get_impl();
```
This seems like overkill, though you could have the non-const getter simply call the const getter.